src/Microsoft.DotNet.Interactive.Http.Tests/ParserTests.Request.cs

```diff
@@ -263,5 +263,66 @@ public void binding_for_variable_in_header_is_correct()
             bindingResult.Value.RequestUri.ToString().Should().Be("https://httpbin.org/anything");
             bindingResult.Value.Headers.ToString().Should().Be("");
         }
+
+        [Fact]
+        public void comments_with_comment_separator_nodes_and_requests()
+        {
+            var result = Parse(
+                """
+                ### Slow Response (Json)
+                GET https://httpbin.org/anything?page=2&pageSize=10
+                ###
+
+                ### Large Response (Json)
+                GET https://dotnetnew-api.azurewebsites.net/api/templatepack
+                ###
+
+                ### Test
+                GET https://www.microsoft.com
+                ###
+
+                ### Content header without content
+                GET https://httpbin.org/anything
+                Content-Type:text/json
+                Content-Length:100
+
+                ###
+
+
+                ###
+                @host = https://www.microsoft.com
+                POST {{host}}
+                ###
+
+                ### Nonexistent
+                GET https://www.sdsdfijj.com
+                ###
+
+                ### Encoding
+                POST http://httpbin.org/anything/anything?id=abáéí?ú?óüÁÉÍ?Ú?ÓÜ
+
+                {
+                  "id":"abáéíőúűóüÁÉÍŐÚŰÓÜ"
+                }
+                ###
+
+                GET https://httpbin.org/anything
+
+                name=value
+
+                ####
+
+                GET https://httpbin.org/anything
+
+                {name=value}
+                """
+                );
```

Comment:
I think it may be better to create a more targeted test for the scenario being fixed.

Reply:
Namely less requests and intricacies?

Reply:
Yes, a tight test that tests exactly the single case that was not working before and that is fixed now.